Tests/Tests.Actors.BugFinding/GC/FinalizerTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Coyote.Runtime;
+using Microsoft.Coyote.SystematicTesting;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Coyote.Actors.BugFinding.Tests
+{
+    public class FinalizerTests : BaseActorBugFindingTest
+    {
+        public FinalizerTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        private class GCTracker
+        {
+            internal bool IsFinalized;
+        }
+
+        private class SetupEvent : Event
+        {
+            internal readonly GCTracker Tracker;
+
+            internal SetupEvent(GCTracker tracker)
+            {
+                this.Tracker = tracker;
+            }
+        }
+
+        public class A : Actor
+        {
+            private GCTracker Tracker;
+
+            protected override Task OnInitializeAsync(Event initialEvent)
+            {
+                this.Tracker = (initialEvent as SetupEvent).Tracker;
+                return Task.CompletedTask;
+            }
+
+            ~A()
+            {
+                this.Tracker.IsFinalized = true;
+            }
+        }
+
+        [Fact(Timeout = 5000)]
+        public void TestActorFinalizerInvoked()
+        {
+            var tracker = new GCTracker();
+
+            var config = this.GetConfiguration().WithTestingIterations(2);
+            TestingEngine engine = TestingEngine.Create(config, (IActorRuntime r) =>
```

Comment:
@lovettchris I tweaked a bit the unit tests to make sure the `TestingEngine` does not go out of scope before we do the GC check. Else the leak might get trivially resolved because once the whole `TestingEngine` goes away the leak also disappears (but this can hide the issue, which happens between iterations).